packages/hardhat-core/src/internal/hardhat-network/provider/fork/ForkStateManager.ts

```diff
@@ -82,7 +81,7 @@ export class ForkStateManager implements PStateManager {
 
     for (const [index, { address, account }] of accounts.entries()) {
       const nonce = nonces[index];
-      account.nonce = toBuffer(nonce);
+      account.nonce = nonce;
```

Comment:
No, account's never frozen... will open an issue. :)

https://github.com/ethereumjs/ethereumjs-monorepo/blob/156664c918842995757894a46e2ea0b643af3399/packages/util/src/account.ts#L61-L73

Reply:
When I was creating the `Account` class I left the decision out to freeze it in the constructor since I noticed the way it was used frequently in the vm was to increment the nonce or change the balance and update it in the state, I didn't think it was necessary to create a new object every time.

Looking at it now, I think I could add a last optional param called `opts` with a `{ freeze }` property like we do in our monorepo classes. That said, if it was important to the end user that Account be immutable, they could just freeze it themselves after instantiation and then I wouldn't have to add this code hehe.